### PR TITLE
Rebalance support

### PR DESCRIPTION
**Pull request by jcustenborder**

### CODE REVIEW

File: src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright © 2016 Jeremy Custenborder (jcustenborder@gmail.com)
```
Comment (sdrichter) [inline src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java]: Just glancing through this and noticed the new files have a copyright year of 2016. Might want to change that :)

File: src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    this.errorPath = errorPath;
+    this.finishedPath = finishedPath;
+  }
+
+
+  public static AbstractCleanUpPolicy create(AbstractSourceConnectorConfig config, InputFile inputFile) throws IOException {
+    final AbstractCleanUpPolicy result;
+    switch (config.cleanupPolicy) {
+      case MOVE:
+        result = new Move(inputFile.inputFile, config.errorPath, config.finishedPath);
```
Comment (sdrichter) [inline src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java]: I am not seeing Move in the MR

File: src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        break;
+      case DELETE:
+        result = new Delete(inputFile.inputFile, config.errorPath, config.finishedPath);
```
Comment (sdrichter) [inline src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java]: Also not seeing Delete in the MR

File: src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        break;
+      case NONE:
+        result = new AbstractCleanUpPolicy.None(inputFile.inputFile, config.errorPath, config.finishedPath);
```
Comment (jmitchellwf) [inline src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java]: Can this be `result = new None(...`, like the Move and Delete?

Comment (jcustenborder) [reply src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java]: Thanks! Corrected.

**Pull request merged**